verena/hashServer.go

```diff
@@ -1,136 +1,117 @@
-	Hash      string `json:"hash"`
-	Version   int    `json:"version"`
-	PublicKey string `json:"publicKey"`
+    Hash          string `json:"hash"`
+    Version       int    `json:"version"`
+    PublicKey     string `json:"publicKey"`
 }
 
 func (e Entry) String() string {
-	return fmt.Sprintf("[%s, %d, %s, %s, %d, %s]", e.Hash, e.Version, e.PublicKey, e.Hash_old, e.Version_old, e.PublicKey_old)
-}
-
-func (e Value) String() string {
-	return fmt.Sprintf("[%s, %d, %s]", e.Hash, e.Version, e.PublicKey)
+    return fmt.Sprintf("[%s, %d, %s]", e.Hash, e.Version, e.PublicKey)
 }
 
 type GetRequest struct {
-	Uuid string `json:"uuid"`
+    Uuid string `json:"uuid"`
 }
 
 type PutRequest struct {
-	Uuid  string `json:"uuid"`
-	Entry Entry  `json:"entry"`
+    Uuid  string `json:"uuid"`
+    Entry Entry  `json:"entry"`
+    OldEntry Entry `json:"oldEntry"`
 }
 
-var kv_store = make(map[string]Value)
-
-func add_value_into_map(e Entry, uuid string) {
-	var v Value
-	v.Hash = e.Hash
-	v.PublicKey = e.PublicKey
-	v.Version = e.Version
-	kv_store[uuid] = v
-	fmt.Println("Successful put for uuid: " + uuid + " (entry: " + v.String() + ")")
-}
+var kv_store = make(map[string]Entry)
 
 func get(w http.ResponseWriter, req *http.Request) {
-	var jsonData GetRequest
-	err := json.NewDecoder(req.Body).Decode(&jsonData)
-	if err != nil {
-		panic(err)
-	}
-
-	uuid := jsonData.Uuid
-	kv_value, temp := kv_store[uuid]
-
-	if temp {
-		fmt.Fprintf(w, "get succeeded for uuid:"+uuid+"\n"+uuid+" "+kv_value.String())
-		fmt.Println("Successful get for uuid: " + uuid)
-	} else {
-		fmt.Fprintf(w, "get failed:uuid "+uuid+" does not exist")
-		fmt.Println("failed get for uuid: " + uuid)
-	}
+    var jsonData GetRequest
+    err := json.NewDecoder(req.Body).Decode(&jsonData)
+    if err != nil {
+        panic(err)
+    }
+
+    uuid := jsonData.Uuid
+    value, exists := kv_store[uuid]
+
+    if ! exists {
+        fmt.Println("no key found: " + uuid)
+        fmt.Fprintf(w, "null\n")
+        return
+    }
+    fmt.Fprintf(w, value.String() + "\n")
+    fmt.Println("Successful get for uuid: " + uuid)
 }
 
 func put(w http.ResponseWriter, req *http.Request) {
-	var jsonData PutRequest
-	err := json.NewDecoder(req.Body).Decode(&jsonData)
-	if err != nil {
-		panic(err)
-	}
-
-	uuid := jsonData.Uuid
-	kv_value, temp := kv_store[uuid]
-
-	/* The below code checks
-	 * Hash value corresponding to a uuid in the hash_server matches the old_hash value sent by client.
-	 * Version corresponding to a uuid in the hash_server matches the old_version value sent by client.
-	 * Public key corresponding to a uuid in the hash_server matches the old_publicKey value sent by client.
-	 * Client sends a new version (usually an incrementing counter) for every put request .
-	 * If the document is saved for the first time, it's version should be set to 1.
-	 * Only when the above checks passes the put request is accepted.
-	 * Else the put request is rejected inlieu of replay attack.
-	 */
```

Comment:
this is a very helpful comment to have while implementing the code! however, we don't want to repeat it here if the code itself describes what is happening (in this case, a branching set of `if` statements)